tests/sentry/issues/auto_source_code_config/test_process_event.py

```diff
@@ -48,37 +71,36 @@ def create_event(self, frames: Sequence[Mapping[str, str | bool]], platform: str
         # XXX: In the future fix store_event to return the correct type
         return cast(GroupEvent, self.store_event(data=test_data, project_id=self.project.id))
 
-    def _get_trees_for_org(self, files: Sequence[str]) -> dict[str, RepoTree]:
-        return {"test-org/repo": RepoTree(RepoAndBranch("test-org/repo", "master"), files)}
-
-    def _repo_tree_files(self, files: Sequence[str]) -> list[dict[str, Any]]:
-        return [{"path": file, "type": "blob"} for file in files]
-
-    def _repo_info(self) -> dict[str, str]:
-        return {"full_name": "test-org/repo", "default_branch": "master"}
-
     def _process_and_assert_code_mapping(
         self,
         *,  # Force keyword arguments
-        repo_files: Sequence[str],
+        repo_trees: Mapping[str, Sequence[str]],
         frames: Sequence[Mapping[str, str | bool]],
         platform: str,
         expected_stack_root: str,
         expected_source_root: str,
+        expected_repo_name: str = REPO1,
+        expected_num_code_mappings: int = 1,
     ) -> None:
         with (
-            patch(f"{CLIENT}.get_tree", return_value=self._repo_tree_files(repo_files)),
+            patch(f"{CLIENT}.get_tree", side_effect=create_mock_get_tree(repo_trees)),
             patch(f"{CLIENT}.get_remaining_api_requests", return_value=500),
-            patch(REPO_TREES_GET_REPOS, return_value=[self._repo_info()]),
+            patch(
+                f"{REPO_TREES_INTEGRATION}._populate_repositories",
+                return_value=mock_populate_repositories(),
+            ),
             patch("sentry.utils.metrics.incr") as mock_incr,
         ):
             event = self.create_event(frames, platform)
             process_event(self.project.id, event.group_id, event.event_id)
             code_mappings = RepositoryProjectPathConfig.objects.all()
-            assert len(code_mappings) == 1
-            code_mapping = code_mappings[0]
-            assert code_mapping.stack_root == expected_stack_root
-            assert code_mapping.source_root == expected_source_root
+            assert len(code_mappings) == expected_num_code_mappings
+            code_mapping = code_mappings.filter(
+                stack_root=expected_stack_root,
+                source_root=expected_source_root,
+            ).first()
+            assert code_mapping is not None
+            assert code_mapping.repository.name == expected_repo_name
```

Comment:
We now also check that the code mapping is associated to the correct repository.